sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/ThreadBuilder.java

```diff
@@ -70,9 +83,32 @@ void scheduleReassignmentToUserOnDeadline(
      * @param userId is the user id to assign it to.
      * @return a NodeOutput.
      */
-    public UserTaskOutput assignUserTaskToUser(
+    UserTaskOutput assignUserTaskToUser(String userTaskDefName, WfRunVariable userId);
+
+    /**
+     * Adds a User Task Node, and assigns it to a specific user
+     * @param userTaskDefName is the UserTaskDef to assign.
+     * @param userId is the user id to assign it to.
+     * @param userGroup  is the User's group
+     * @return a NodeOutput.
+     */
+    UserTaskOutput assignUserTaskToUser(
+        String userTaskDefName,
+        WfRunVariable userId,
+        String userGroup
+    );
+
+    /**
+     * Adds a User Task Node, and assigns it to a specific user
+     * @param userTaskDefName is the UserTaskDef to assign.
+     * @param userId is the user id to assign it to.
+     * @param userGroup is the User's group
+     * @return a NodeOutput.
+     */
+    UserTaskOutput assignUserTaskToUser(
```

Comment:
Wait a minute. Is it possible to have a method without a modifier in Java? This is magic! haha